cdap-ui/app/cdap/components/DataPrepConnections/ConnectionPopover/index.js

```diff
@@ -215,19 +215,21 @@ export default class ConnectionPopover extends Component {
   render() {
     return (
       <span className="expanded-menu-popover-icon text-center float-right">
-        <UncontrolledPopover icon="fa-ellipsis-v" popoverClassName="connection-action-popover">
-          <div className="connection-action-item" onClick={this.toggleEdit}>
-            <span>{T.translate(`${PREFIX}.edit`)}</span>
-          </div>
-
-          <div className="connection-action-item" onClick={this.toggleDuplicate}>
-            <span>{T.translate(`${PREFIX}.duplicate`)}</span>
-          </div>
-
-          <div className="connection-action-item" onClick={this.toggleDeleteConfirmation}>
-            <span>{T.translate(`${PREFIX}.delete`)}</span>
-          </div>
-        </UncontrolledPopover>
+        <If condition={!this.props.connectionInfo.builtIn}>
```

Comment:
nit: Wouldn't it be simpler if 
```
if(!this.props.connectionInfo.builtIn) {
 return null;
}
```

Reply:
I actually need the span container so that the white styling still gets applied when the connection is active. At this stage I didn't want to refactor too much, that's why keeping it here.